core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -851,6 +851,26 @@ object TestUtils extends Logging {
     throw new RuntimeException("unexpected error")
   }
 
+  /**
+    * Wait until the given predicate is true or the provided wait time elapses. Return the last value returned by
+    * `compute` and a boolean indicating whether `predicate` succeeded for that value. This method is useful in cases
+    * where `waitUntilTrue` makes it awkward to produce good error messages.
+    */
+  def waitAndGet[T](compute: => T, waitTime: Long = JTestUtils.DEFAULT_MAX_WAIT_MS, pause: Long = 100L)(
```

Comment:
Any suggestions for a better name? I couldn't think of something that was not too long and clear at the same time.